dashboard/app/controllers/reference_guides_controller.rb

```diff
@@ -0,0 +1,15 @@
+class ReferenceGuidesController < ApplicationController
+  include CurriculumHelper
+  authorize_resource
+
+  before_action :require_levelbuilder_mode_or_test_env
+
+  def show
+    if params[:course_course_name] && params[:key]
+      course_version_id = find_matching_course_version(params[:course_course_name])&.id
+      @reference_guide = ReferenceGuide.find_by_course_version_id_and_key(course_version_id, params[:key])
+      return render :show if @reference_guide
```

Comment:
I think if you do the above error checking along the way you don't need this and the following lines. Rails does some magic to automatically call render :show without you needing to write it